frontend/src/main/scala/bloop/cli/Proxy.scala

```diff
@@ -0,0 +1,90 @@
+package bloop.cli
+
+import java.net.URL
+import java.net.MalformedURLException
+
+import scala.util.control.NonFatal
+
+import bloop.engine.State
+
+/**
+  * Proxy helper module
+  * Information came from  https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html
+  */
+object Proxy {
+  /**
+    * Extract environments variables from the given state to set the proxy properties of the jvm.
+    * If environement variable are not found in the state, the property corresponding is cleared,
+    * to let user add and remove env var.
+    * */
+  def setProxyJvmSettings(state : State) : Unit = {
+    //no_proxy environment variable seems not be binded to a protocol.
+
+    val maybeNoProxy = Option(state.commonOptions.env.getProperty("no_proxy")).map(_.replace(',','|'))
+
+    val maybeHttpProxy = state.commonOptions.env.getProperty("http_proxy")
+    if(maybeHttpProxy != null) {
+      try {
+        val url =  new URL(maybeHttpProxy)
+        System.setProperty("http.proxyHost",url.getHost())
+        System.setProperty("http.proxyPort",url.getPort().toString)
+        maybeNoProxy.foreach(System.setProperty("http.nonProxyHosts", _))
+      } catch {
+        case e : MalformedURLException => //Ignore, but should maybe log a warning if the URL is not parsable.
```

Comment:
Let's debug a message if this exception happens, such as "Unexpected non-URI format in proxy value of `$key`"?